shared/constants/signatures.ts

```diff
@@ -0,0 +1,29 @@
+export enum PrimaryTypeOrder {
+  Order = 'Order',
+  OrderComponents = 'OrderComponents',
+}
+
+export enum PrimaryTypePermit {
+  Permit = 'Permit',
+  PermitBatch = 'PermitBatch',
+  PermitBatchTransferFrom = 'PermitBatchTransferFrom',
+  PermitSingle = 'PermitSingle',
+  PermitTransferFrom = 'PermitTransferFrom',
+}
+
+/**
+ * EIP-712 Permit PrimaryTypes
+ */
+export enum PrimaryType {
```

Comment:
Minor, but could we generate this a a union of the above two enums to avoid the duplication?

Reply:
oops, forgot to update this. Thank you! Let me know if there are any other updates here [1c6a6a7](https://github.com/MetaMask/metamask-extension/pull/26107/commits/1c6a6a78fae172c9e3775df957a6b096be389156)